include/xsimd/arch/generic/xsimd_generic_memory.hpp

```diff
@@ -55,6 +55,42 @@ namespace xsimd
             return batch<T, A>::load_aligned(concat_buffer);
         }
 
+        // gather
+        namespace detail
+        {
+            template <typename T, typename A, typename U, typename V, size_t I>
+            inline batch<T, A> gather(U const* src, batch<V, A> const& index,
+                                      ::xsimd::detail::index_sequence<I>) noexcept
+            {
+                return insert(batch<T, A> {}, static_cast<T>(src[index.get(I)]),
+                              ::xsimd::index<I>());
+            }
+
+            template <typename T, typename A, typename U, typename V, size_t I,
+                      size_t... Is,
+                      typename = typename std::enable_if<sizeof...(Is) != 0>::type>
+            inline batch<T, A>
+            gather(U const* src, batch<V, A> const& index,
+                   ::xsimd::detail::index_sequence<I, Is...>) noexcept
+            {
+                const auto test = kernel::detail::gather<T, A, U, V>(
+                    src, index, ::xsimd::detail::index_sequence<Is...>());
+                return insert(test, static_cast<T>(src[index.get(I)]),
+                              ::xsimd::index<I>());
+            }
+        } // namespace detail
+
+        template <typename A, typename T, typename U, typename V>
+        inline batch<T, A> gather(U const* src, batch<V, A> const& index,
+                                  kernel::requires_arch<generic>) noexcept
+        {
+            static_assert(batch<T, A>::size <= batch<V, A>::size,
```

Comment:
why `<=` and not `==` ?

Reply:
Vc makes it possible to use `int8_t` or `int16_t` indexes with a `float*` or `double*`, so I did not want to restrain user choice here. But do let me know if you want it restricted here.

Reply:
Well, you're checking the `size` so the number of elements, I think they *must* match.